### PR TITLE
app-misc/g810-led: New Package

### PR DESCRIPTION
**Pull request by lanodan**
Based on ::zyrenth

It’s still using EAPI 6 because the udev and eutils eclass aren’t marked as
compatible with EAPI 7 (yet?).

Closes: https://bugs.gentoo.org/662494

EDIT: Updated the commit message for EAPI 7 systemd but non-EAPI 7 udev and eutils.

### CODE REVIEW

File: app-misc/g810-led/g810-led-0.2.7.ebuild [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	emake LIB="${my_lib}" || die "emake failed"
+}
+
+src_install() {
+	dolib.so "lib/libg810-led.so.${PV}"
+	dosym "libg810-led.so.${PV}" "usr/$(get_libdir)/libg810-led.so"
+
+	dobin bin/g810-led
+	boards=(213 410 413 610 910 pro)
+	for x in ${boards[@]}; do
```
Comment (jonasstein) [inline app-misc/g810-led/g810-led-0.2.7.ebuild]: local x is still missing

File: app-misc/g810-led/g810-led-0.2.7.ebuild [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+src_compile() {
+	local my_lib=""
+	if use hidapi; then
+		my_lib="hidapi"
+	else
+		my_lib="libusb"
+	fi
+
+	emake LIB="${my_lib}" || die "emake failed"
```
Comment (monsieurp) [inline app-misc/g810-led/g810-led-0.2.7.ebuild]: `die` is not needed here. Please remove it.

File: app-misc/g810-led/g810-led-0.2.7.ebuild [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+		dosym g810-led "/usr/bin/g${x}-led"
+	done
+
+	systemd_dounit systemd/g810-led.service
+	systemd_dounit systemd/g810-led-reboot.service
+
+	insinto /lib/udev/rules.d/
+	newins udev/g810-led.rules 60-g810-led.rules
+
+	dodoc README.md
```
Comment (monsieurp) [inline app-misc/g810-led/g810-led-0.2.7.ebuild]: call `einstalldocs` here please.

File: app-misc/g810-led/g810-led-0.2.7.ebuild [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+REQUIRED_USE="^^ ( hidapi libusb )"
+
+RDEPEND="
+	hidapi? ( dev-libs/hidapi:= )
+	libusb? ( virtual/libusb:= )
+"
+
+DEPEND="${RDEPEND}"
+
+DOCS="README.md sample_profiles"
```
Comment (mgorny) [inline app-misc/g810-led/g810-led-0.2.7.ebuild]: Array, please.

File: app-misc/g810-led/g810-led-0.2.7.ebuild [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+src_prepare() {
+	sed -i -e 's/ \/etc\// \$(DESTDIR)\/etc\//' makefile || die "Sed failed!"
```
Comment (mgorny) [inline app-misc/g810-led/g810-led-0.2.7.ebuild]: You aren't really using the install rules, are you?

File: app-misc/g810-led/g810-led-0.2.7.ebuild [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+src_compile() {
+	local my_lib=""
+	if use hidapi; then
+		my_lib="hidapi"
+	else
+		my_lib="libusb"
+	fi
+
+	emake LIB="${my_lib}"
```
Comment (mgorny) [inline app-misc/g810-led/g810-led-0.2.7.ebuild]: Use `usex` for this.

File: app-misc/g810-led/g810-led-0.2.7.ebuild [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+src_install() {
+	dolib.so "lib/libg810-led.so.${PV}"
+	dosym "libg810-led.so.${PV}" "usr/$(get_libdir)/libg810-led.so"
```
Comment (mgorny) [inline app-misc/g810-led/g810-led-0.2.7.ebuild]: Please start the path on RHS with a slash.

File: app-misc/g810-led/g810-led-0.2.7.ebuild [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+	dobin bin/g810-led
+	boards=(213 410 413 610 910 pro)
```
Comment (mgorny) [inline app-misc/g810-led/g810-led-0.2.7.ebuild]: This var also needs `local`.

File: app-misc/g810-led/g810-led-0.2.7.ebuild [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+	local x
+	for x in "${boards[@]}"; do
+		dosym g810-led "/usr/bin/g${x}-led"
+	done
+
+	systemd_dounit systemd/g810-led.service
+	systemd_dounit systemd/g810-led-reboot.service
+
+	insinto /lib/udev/rules.d/
+	newins udev/g810-led.rules 60-g810-led.rules
```
Comment (mgorny) [inline app-misc/g810-led/g810-led-0.2.7.ebuild]: `udev_newrules` (udev.eclass)?

**Comment by lanodan** [comment]
I fixed the environment issue by patching the makefile.

File: app-misc/g810-led/files/g810-led-0.2.7-Have_makefile_use_environment.patch [hunk 10]
```diff
@@ -0,0 +1,62 @@
+diff --git a/makefile b/makefile
+--- a/makefile
++++ b/makefile
+@@ -1,18 +1,18 @@
+-CC=g++
```
[the remark this answers is not in the document]
Comment (lanodan) [reply app-misc/g810-led/files/g810-led-0.2.7-Have_makefile_use_environment.patch]: I planned to submit it upstream ( https://github.com/MatMoul/g810-led/pull/139 ), and I forgot the overriding via arguments.

File: app-misc/g810-led/files/g810-led-0.2.7-Have_makefile_use_environment.patch [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+-	LDFLAGS=-lusb-1.0
++	LDFLAGS+=-lusb-1.0
+ else
+ 	CPPFLAGS=-Dhidapi
+-	LDFLAGS=-lhidapi-hidraw
++	LDFLAGS+=-lhidapi-hidraw
+ endif
+ SYSTEMDDIR?=/usr/lib/systemd
+ 
+-prefix?=$(DESTDIR)/usr
```
[the remark this answers is not in the document]
Comment (lanodan) [reply app-misc/g810-led/files/g810-led-0.2.7-Have_makefile_use_environment.patch]: Isn’t the capitalised one the ~standard one?

[the remark this answers is not in the document]
Comment (mgorny) [reply app-misc/g810-led/files/g810-led-0.2.7-Have_makefile_use_environment.patch]: I don't think there is a standard there. I think autotools are actually using lowercase there.

**Pull request closed without merging**